src/app/shared/components/button/button.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component } from '@angular/core';
-import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
-import { fas } from '@fortawesome/free-solid-svg-icons';
+import { Component, Input } from "@angular/core";
+import { FaIconLibrary } from "@fortawesome/angular-fontawesome";
+import { fas, IconName } from "@fortawesome/free-solid-svg-icons";
 
 @Component({
-  selector: 'app-button',
-  templateUrl: './button.component.html',
-  styleUrls: ['./button.component.scss']
+  selector: "app-button",
+  templateUrl: "./button.component.html",
+  styleUrls: ["./button.component.scss"],
 })
 export class ButtonComponent {
+  private library: FaIconLibrary;
+
   constructor(library: FaIconLibrary) {
```

Comment:
As you add `FaIconLibrary` as an argument in the constructor, then it will be available in you component during it's lifecycle. You don't have to assign it to another variable.
By marking the visibility of the constructor argument, it increase readability and intent. In this case visibility can be set to `private` since it will not be accessed from outside. After that you can safely call `this.library` without the need to assign it to another library instance.